doc/en/fixture.rst

```diff
@@ -492,12 +492,16 @@ with ``--collect-only`` will show the generated IDs.
 Numbers, strings, booleans and None will have their usual string
 representation used in the test ID. For other objects, pytest will
 make a string based on the argument name.  It is possible to customise
-the string used in a test ID for a certain fixture value by using the
-``ids`` keyword argument::
+the string used in a test ID for a certain fixture value, by declaring an attribute
+`pytest_id` in the object itself, or by using the ``ids`` keyword argument::
 
    # content of test_ids.py
    import pytest
 
+   class Cheese(object):
+       def __init__(self, name)
+           self.pytest_id = name
```

Comment:
Not sure about the name, as a constant perhaps it should be `PYTEST_ID`?

Reply:
@RonnyPfannschmidt @The-Compiler what do you guys think?

Reply:
at first glance, this is a general no go - this is pretty much a hidden structual dependency